Packages/com.chisel.editor/Chisel/Editor/Editor/Scene/CSGSceneBottomGUI.cs

```diff
@@ -13,670 +13,98 @@ namespace Chisel.Editors
-		const float kButtonWidth        = 18;
-		const float kDropdownWidth      = kButtonWidth + 8;
+		const float kBottomBarHeight    = 20;
 		const float kTopBarHeight       = 17;
 		const float kFloatFieldWidth    = 60;
 
 		static readonly int BottomBarGUIHash = typeof(CSGSceneBottomGUI).Name.GetHashCode();
-		static GUIStyle miniLabel;
-		static GUIStyle lockXButtonStyle;
-		static GUIStyle lockYButtonStyle;
-		static GUIStyle lockZButtonStyle;
-		static Color[]  lockButtonToggleColor;
-		static bool     prevSkin = false;
-		static GUIContent rebuildContent;
-		static GUIContent viewContent;
-
-		static readonly GUIContent lockContent              = new GUIContent("Lock");   
-		static readonly GUIContent lockXAxisContent         = new GUIContent("X");
-		static readonly GUIContent lockYAxisContent         = new GUIContent("Y");
-		static readonly GUIContent lockZAxisContent         = new GUIContent("Z");
-
-		static readonly GUIContent boundsSnappingContent	= new GUIContent("Snap-Bounds");
-		static readonly GUIContent pivotSnappingContent		= new GUIContent("Snap-Pivot");
-		static readonly GUIContent rotationSnappingContent  = new GUIContent("Snap-Rotate");
-		static readonly GUIContent scaleSnappingContent     = new GUIContent("Snap-Scale");
 
-		static readonly GUIContent boundsContent			= new GUIContent("Bounds");
-		static readonly GUIContent pivotContent				= new GUIContent("Pivot");
-		static readonly GUIContent rotationContent          = new GUIContent("Rotate");
-		static readonly GUIContent scaleContent             = new GUIContent("Scale");
-
-		static readonly GUIContent allAxiContent            = new GUIContent("XYZ"); 
-		static readonly GUIContent xAxisContent             = new GUIContent("x");
-		static readonly GUIContent yAxisContent             = new GUIContent("y");
-		static readonly GUIContent zAxisContent             = new GUIContent("y");
-
-		static readonly GUIContent increaseContent          = new GUIContent("+");
-		static readonly GUIContent decreaseContent          = new GUIContent("-");
+		static GUILayoutOption floatWidthLayout;
 
+		static GUIStyle toolbarStyle;
+		static GUIStyle toggleStyle;
+		static GUIStyle buttonStyle;
+		static bool     prevSkin = false;
 
-		// TODO: put somewhere else
-		// TODO: use "Editor Default Resources" directory?
-		// TODO: make this a scriptable object and just set defaults on script?
-		static GUIContent IconContent(string name)
-		{
-            var path = "Assets/Plugins/Chisel/PluginAssets/Icons/" + name + ".png";
-			var image = AssetDatabase.LoadAssetAtPath<Texture2D>(path);
-			return new GUIContent(image);
-		}
+		static readonly int bottomBarGuiId					= 21001;
 
 
 		public static void Rebuild()
 		{
 			CSGNodeHierarchyManager.Rebuild();
 		}
 
-		public static void CycleDistanceUnits()
-		{
-			CSGEditorSettings.Load();
-			CSGEditorSettings.DistanceUnit = Units.CycleToNextUnit(CSGEditorSettings.DistanceUnit);
-			CSGEditorSettings.Save();
-		}
```

Comment:
having units on the toolbar makes it easier for people to reason about numbers, the same units would be used in the scene view to show how large something is

Reply:
I'm not really convinced about this? Unless you adhere strictly from the start the distance units will likely never make sense for your project, making them worse than useless for most projects

Reply:
But how can you adhere to using actual units if your tools don't support it? If you know that something is a certain amount of meters or inches, it makes more intuitive sense to you .. if you make something 2 units tall and pretend it's a building .. then you might be unpleasantly surprised that unity's physics system assumes 1 unit is 1 meter.. on the other hand, if you want to make something that has realistic dimensions then you know that a 2m high door is reasonable

Reply:
Ah I didn't know that about the physics system, that's an interesting wrinkle. In terms of the units themselves the numbers still exist and are a point of reference for the user, having a real-world measurement doesn't necessarily make that easier.

Thinking about it more, especially with that knowledge of the physics system (and potentially other plugins people may be using), it makes sense now. We could have a customizable units thing in settings somewhere at a later date.